src/org/stringtemplate/v4/Interpreter.java

```diff
@@ -793,7 +793,7 @@ protected int writePOJO(STWriter out, InstanceScope scope, Object o, String[] op
         String formatString = null;
         if ( options!=null ) formatString = options[Option.FORMAT.ordinal()];
         // ask the native group defining the surrounding template for the renderer
-        AttributeRenderer r = scope.st.impl.nativeGroup.getAttributeRenderer(o.getClass());
+        AttributeRenderer<? super Object> r = scope.st.impl.nativeGroup.getAttributeRenderer(o.getClass());
```

Comment:
😟 I'm not a fan of this. It is definitely an inaccurate representation of the types involved. My recommendation is to extract the code that initializes `v` to a new helper method:

```
private <T> String renderObject(InstanceScope scope, String formatString, Object o, Class<T> attributeType) {
	AttributeRenderer<? super T> r = scope.st.impl.nativeGroup.getAttributeRenderer(attributeType);
	if ( r!=null ) {
		return r.toString(attributeType.cast(o), formatString, locale);
	} else {
		return o.toString();
	}
}
```

Then update this method to define and initialize `v` with the following:

```
String v = renderObject(scope, formatString, o, o.getClass());
```

Reply:
Addressed by 7d870fae2f5c38180e3333228cc903a4d83a651c